### PR TITLE
[SPARK-8618] Obtain hbase token retries many times when having hbase class but no hbase configuration

### PR DESCRIPTION
**Pull request by XuTingjun**

### CODE REVIEW

File: yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala [hunk 1]
```diff
@@ -1258,10 +1258,17 @@ object Client extends Logging {
...
 
         val hbaseConf = confCreate.invoke(null, conf)
-        val token = obtainToken.invoke(null, hbaseConf).asInstanceOf[Token[TokenIdentifier]]
-        credentials.addToken(token.getService, token)
-
-        logInfo("Added HBase security token to credentials.")
+        val hbaseConfGet = (param: String) => Option(confClass
+          .getMethod("get", classOf[java.lang.String])
+          .invoke(hbaseConf, param))
+        val zk_quorum = hbaseConfGet("hbase.zookeeper.quorum")
```
Comment (srowen) [inline yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala]: Fix this variable name - zkQuorum

**Comment by srowen** [comment]
Can you improve the title of this PR? it doesn't relate to the JIRA

**Comment by sujkh85** [comment]
NAVER - http://www.naver.com/
--------------------------------------------

sujkh@naver.com 님께 보내신 메일 <Re: [spark] [SPARK-8618] Obtain hbase token retries many times when having hbase class but no hbase configuration (#7007)> 이 다음과 같은 이유로 전송 실패했습니다.

--------------------------------------------

받는 사람이 회원님의 메일을 수신차단 하였습니다. 


--------------------------------------------

**Comment by tgravescs** [comment]
What is the use case for adding hbase jars but not the configuration?

File: yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala [hunk 2]
```diff
@@ -1258,10 +1258,17 @@ object Client extends Logging {
...
 
         val hbaseConf = confCreate.invoke(null, conf)
-        val token = obtainToken.invoke(null, hbaseConf).asInstanceOf[Token[TokenIdentifier]]
-        credentials.addToken(token.getService, token)
-
-        logInfo("Added HBase security token to credentials.")
+        val hbaseConfGet = (param: String) => Option(confClass
+          .getMethod("get", classOf[java.lang.String])
+          .invoke(hbaseConf, param))
+        val zkQuorum = hbaseConfGet("hbase.zookeeper.quorum")
```
Comment (tgravescs) [inline yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala]: Any particular reason to the zookeeper.quorum conf here?  I'm guessing you just chose one to try.

**Comment by andrewor14** [comment]
@XuTingjun please add `[YARN]` to the title, thanks.

**Comment by XuTingjun** [comment]
sorry, I think this patch is not good, so i will close it.

**Pull request closed without merging**